src/ComboBox/ComboBox.svelte

```diff
@@ -299,11 +298,26 @@
                 selectedId = filteredItems[highlightedIndex].id;
               }
             } else {
-              open = false;
-              if (filteredItems[0]) {
-                value = itemToString(filteredItems[0]);
-                selectedItem = filteredItems[0];
-                selectedId = filteredItems[0].id;
+              // searching typed value in text list with lowercase
+              const matchedItem = filteredItems.find(
+                (e) =>
+                  e.text.toLowerCase() === value?.toLowerCase() && !e.disabled
+              );
+              if (matchedItem) {
+                // typed value has matched
+                open = false;
+                selectedItem = matchedItem;
+                value = selectedItem.text;
+                selectedId = selectedItem.id;
+              } else {
+                open = false;
+                // find first enabled item
+                const matchedItem = filteredItems.find((e) => !e.disabled);
+                if (matchedItem) {
+                  value = itemToString(matchedItem);
+                  selectedItem = matchedItem;
+                  selectedId = matchedItem.id;
+                }
```

Comment:
This looks like it could be simplified further, the state updates look like they should be identical.

There is a discrepancy with the `value`:
```js
// if
value = selectedItem.text;
// else
value = itemToString(matchedItem);
```

Maybe set `matchedItem` in way that directly falls back to the first item, then the `else` could maybe be omitted?

Something like

```js
const matchedItem =
  filteredItems.find((e) => !e.disabled && e.text.toLowerCase() === value?.toLowerCase()) ??
  filteredItems.find((e) => !e.disabled);
```

(Though `prettier` will probably make this look ugly 😅)

You could also pre-filter on the `disabled`, since this is required in any case. Not sure if this would negatively impact performance for large lists, though. (Ideally one would use a [generator function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Generator) so the list is not iterated yet, but their ergonomics are currently not that great. [This proposal](https://github.com/tc39/proposal-iterator-helpers) would fix that.)

Reply:
OK, I think it can be done. Thanks!

Reply:
Isn't `itemToString` the correct version of setting the `value`?
(That is what was used prior in the code.)

Reply:
Hm. Good question. Based on the code, it seems that itemToString is useful, if the text is empty. On the other hand, according to the definition of the type, "text" cannot be empty.
```javascript
@typedef {{ id: ComboBoxItemId; text: string; disabled?: boolean; }} ComboBoxItem
```
I will modify it according to your suggestion.